cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -286,4 +305,48 @@ llvm::Value* LLVMJITValue::store(LLVMJITValue& rh) {
   return nullptr;
 }
 
+JITValuePointer LLVMJITValue::castPointerSubType(JITTypeTag type_tag) {
+  if (JITTypeTag::POINTER != getValueTypeTag()) {
+    LOG(FATAL) << "Invalid operation to cast subtype of a non-POINTER JITValue. Type="
+               << getJITTypeName(getValueTypeTag());
+  }
+  llvm::Value* new_llvm_value =
+      getFunctionBuilder(parent_function_)
+          .CreateBitCast(load(),
+                         getLLVMPtrType(type_tag, parent_function_.getLLVMContext()));
+  return std::make_unique<LLVMJITValue>(JITTypeTag::POINTER,
+                                        parent_function_,
+                                        new_llvm_value,
+                                        "cast_ptr",
+                                        JITBackendTag::LLVMJIT,
+                                        false,
+                                        type_tag);
+}
+
+JITValuePointer LLVMJITValue::dereference() {
+  if (JITTypeTag::POINTER != getValueTypeTag()) {
+    LOG(FATAL) << "Invalid operation to dereference of a non-POINTER JITValue. Type="
+               << getJITTypeName(getValueTypeTag());
+  }
+
+  return std::make_unique<LLVMJITValue>(getValueSubTypeTag(),
+                                        parent_function_,
+                                        llvm_value_,
+                                        "dereference_ptr",
+                                        JITBackendTag::LLVMJIT,
+                                        true,
+                                        JITTypeTag::INVALID);
+}
+
+JITValuePointer LLVMJITValue::getElemAt(JITValue& index) {
+  if (JITTypeTag::POINTER != getValueTypeTag()) {
```

Comment:
seems we mix `pointer of type` and `array of type`?

Reply:
Yes